assignment_grade.rb

```diff
@@ -1,5 +1,8 @@
 class AssignmentGrade < ActiveRecord::Base
 
+  belongs_to :assignment
+  belongs_to :student_id, class_name: "CourseStudent", foreign_key: :course_student_id
```

Comment:
Would have been a bit better like:

```ruby
belongs_to :course_student
belongs_to :student, through: :course_student
```

Might have needed a bit more details on the `:student` relationship, but the general idea is `course_student` is little more than a join table.